edi_oca/utils.py

```diff
@@ -2,9 +2,25 @@
 # @author Simone Orsi <simahawk@gmail.com>
 # License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
 
+import hashlib
+
 from odoo.addons.http_routing.models.ir_http import slugify
+from odoo.addons.queue_job.job import identity_exact_hasher
 
 
 def normalize_string(a_string, sep="_"):
     """Normalize given string, replace dashes with given separator."""
     return slugify(a_string).replace("-", sep)
+
+
+def get_checksum(filecontent):
+    return hashlib.md5(filecontent).hexdigest()
+
+
+def exchange_record_job_identity_exact(job_):
+    hasher = identity_exact_hasher(job_)
+    # Include files checksum
+    hasher.update(
+        str(sorted(job_.recordset.mapped("exchange_filechecksum"))).encode("utf-8")
```

Comment:
Because is the only to way to know if the file is the same.
If by error you try to schedule another job you won't get duplicated jobs unless the content of the file is the same.
In most cases, if you rely only on the cron to process files you'll have the filter done by the check methods that will skip records not in the rate state or not w/ the right values to be generated/sent/received/processed.
However, if you force those checks and try to schedule a job for the a record whereas the file didn't change you shouldn't be able to.

Reply:
It might come handy also to compare files w/o reading the whole file (eg: to check if you received the same file twice).

Reply:
What will happen if the file is not generated?